app/models/with_class_info.rb

```diff
@@ -0,0 +1,28 @@
+module WithClassInfo
+
+  # If we have new class_info_url or class_hash in our query params
+  # set our class_hash and class_info URLs to that (if they are blank)
+  # Discussion: Not setting these values if they are already set seems safest.
```

Comment:
There is the case where a student moves from one class to another. We just added support for admins to do that in the Portal: https://github.com/concord-consortium/rigse/pull/653 
This code reuses the same Learner model so it has the same remote_endpoint, which means the same LARA Run or SequenceRun will be found.  

By resetting the class params on each launch, then they will be correct at the time a material is re-launched. However this approach is not going to help the report service which storing this info to reconstruct the report.

So it is probably better to have an explicit 'update' operation which can notify any external systems when something is changed and not need to relaunch.  There doesn't seem to be built in LTI support for this, but it could be a feature of our external Tool as long as we can come up with a way for a user to tell us what the two context_ids and resource_link_ids are.

There is a PT story for adding support to the report service for handling moving students from class to class: https://www.pivotaltracker.com/story/show/166217660 we'll have to deal with this issue when we work on that story.